### DISCUSSION TITLE
Badge on Resource Sub-Navigation

### DISCUSSION THREAD

**Discussion by diogogpinto** [filamentphp/filament#10168] [Help]
Hello, friends 👋🏻

On a Resource Page, how can you get the count for a relationship?

In this example, my method has a a 'notes" relationship... and in the custom page for that resource, I have the following code:

```php
public static function getNavigationBadge(): ?string
{
    return static::getRecord()::notes()::count();
}
```

But I get the following error:

```
Non-static method App\Filament\Extensions\NotesResourcePage::getRecord() cannot be called statically
```

Am I doing something wrong or is this something that can't be done at the moment?

Thank you!

**Comment by diogogpinto** [comment]
This is a workaround I got... not ideal, I suppose, but will wait on you guys' feedback!

```php
public static function getNavigationBadge(): ?string
{
    return self::getResource()::getModel()::findOrFail(request()->route()->parameter('record'))->notes()->count();
}
```

**Reply by leandrocfe** [reply]
You could override the mount method

```php
public static $totalNotes = 0;

public function mount(int|string $record): void
{
    parent::mount($record);
    static::$totalNotes = $this->getRecord()->notes()->count();
}

public static function getNavigationBadge(): ?string
{
    return static::$totalNotes;
}
```

**Reply by diogogpinto** [reply]
Hey @leandrocfe , thank you for you feedback!

That doesn't work. I'm using this on a resource custom page subnavigation. When I use your code, I get:

```
BadMethodCallException
Method App\Filament\Extensions\NotesResourcePage::mount does not exist.
```

I tried the following:

```php
public static $badgeCount = 0;

public function mount(int | string $record): void
{
    $this->record = $this->resolveRecord($record);
    $this->form->fill();
    static::$badgeCount = $this->getRecord()->notes()->count();
    static::authorizeResourceAccess();
}

public static function getNavigationBadge(): ?string
{
    return static::$totalNotes;
}

```

But when I access the resource it shows 0, only when I access the notes it will show the correct number. See attachment below.

<img width="687" src="https://github.com/filamentphp/filament/assets/74505947/b24d16b8-75ce-4b34-8584-18c47f830113">
<img width="677" src="https://github.com/filamentphp/filament/assets/74505947/ce329fef-af77-48e7-bbf5-7bd7c0647899">

**Comment by diogogpinto** [comment]
So far I haven't found any solution to this problem and I don't think there's any other possible solution, because of how the badge is called when browsing other items in the navigation.

The only working solution so far is really:

```php
    public static function getNavigationBadge(): ?string
    {
        return self::getResource()::getModel()::findOrFail(request()->route()->parameter('record'))->notes()->count();
    }
```
I don't know if performance wise o security wise this is safe, though.

**Reply by GamendeBrian** [reply]
In my case when I use headerActions it kept giving errors, so I empty checked it, instead of failing the method.

```php
public static function getNavigationBadge(): ?string
{
    $record = self::getResource()::getModel()::find(request()->route()->parameter('record'));
    return $record ? $record->exports()->count() : 0;
}
```

**Reply by ryanmortier** [reply]
You can achieve the same syntax but with a null safe operator.

```php
    public static function getNavigationBadge(): ?string
    {
        return self::getResource()::getModel()::find(request()->route()->parameter('record'))?->notes()->count();
    }
```

**Comment by erchecho** [comment]
I know this is an old question, but I found this somewhere else, and I think it is a good approach. 

```
public static function getNavigationBadge(): ?string
{
    return \Livewire::current()->getRecord()->notes->count();
}
```

**Comment by kingmaker-agm** [comment]
Thanks for the wonderful solution @erchecho 👍
